common/utils/src/main/scala/org/apache/spark/internal/Logging.scala

```diff
@@ -37,7 +37,10 @@ import org.apache.spark.util.SparkClassUtils
  * The values of the MDC will be inline in the log message, while the key-value pairs will be
  * part of the ThreadContext.
  */
-case class MDC(key: LogKey, value: Any)
+case class MDC(key: LogKey, value: Any) {
+  require(!value.isInstanceOf[MessageWithContext],
```

Comment:
Let's run the check if it is testing.

Reply:
I am concerned that there is a mistake and we didn't capture it before release.

Reply:
Yeah